### PR TITLE
CMR-6020: Added retries to execute ES query on UnknownHostException

### PR DESCRIPTION
**Pull request by ygliuvt**
James and I add this retry to resolve the 503 error related to search encounters UnknownHostException when talking to ES cluster via DNS name. We are planning to backport this fix to PROD.

### CODE REVIEW

File: common-app-lib/src/cmr/common_app/services/search/elastic_search_index.clj [hunk 1]
```diff
@@ -100,12 +101,28 @@
...
+         (context->conn context) (:index-name index-info) [(:type-name index-info)] query))
+      (errors/throw-service-error :service-unavailable "Exhausted retries to execute ES query"))
+    (catch UnknownHostException e
+      (info (format
+             (str "Excute ES query failed due to UnknownHostException. Retry in half a second."
+                  " Will retry up to %d times.")
+             max-retries))
+      (Thread/sleep 500)
+      (do-send-with-retry context index-info query (- max-retries 1)))))
+
```
Comment (mschmele) [inline common-app-lib/src/cmr/common_app/services/search/elastic_search_index.clj]: Is there a way to add a unit test for this function?

File: common-app-lib/src/cmr/common_app/services/search/elastic_search_index.clj [hunk 2]
```diff
@@ -100,12 +101,28 @@
...
+  (try
+    (if (> max-retries 0)
+      (if-let [scroll-id (:scroll-id query)]
+        (scroll-search context scroll-id)
+        (esd/search
+         (context->conn context) (:index-name index-info) [(:type-name index-info)] query))
+      (errors/throw-service-error :service-unavailable "Exhausted retries to execute ES query"))
+    (catch UnknownHostException e
+      (info (format
+             (str "Excute ES query failed due to UnknownHostException. Retry in half a second."
```
Comment (CayvonH) [inline common-app-lib/src/cmr/common_app/services/search/elastic_search_index.clj]: Knit: Execute mispelled.

File: common-app-lib/src/cmr/common_app/services/search/elastic_search_index.clj [hunk 3]
```diff
@@ -100,12 +101,28 @@
...
+                  " Will retry up to %d times.")
+             max-retries))
+      (Thread/sleep 500)
```
Comment (CayvonH) [inline common-app-lib/src/cmr/common_app/services/search/elastic_search_index.clj]: Might be nice to make this sleep time also configurable

File: common-app-lib/src/cmr/common_app/services/search/elastic_search_index.clj (hunk 2 above)
Comment (ygliuvt) [reply common-app-lib/src/cmr/common_app/services/search/elastic_search_index.clj]: Done.

File: common-app-lib/src/cmr/common_app/services/search/elastic_search_index.clj (hunk 3 above)
Comment (ygliuvt) [reply common-app-lib/src/cmr/common_app/services/search/elastic_search_index.clj]: Added

**Pull request merged**